pkg/await/awaiters.go

```diff
@@ -16,15 +16,149 @@ package await
 
 import (
 	"fmt"
+	"reflect"
 	"time"
 
 	"github.com/golang/glog"
+	"github.com/pulumi/pulumi-kubernetes/pkg/client"
+	"github.com/pulumi/pulumi-kubernetes/pkg/openapi"
 	"github.com/pulumi/pulumi-kubernetes/pkg/watcher"
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
 	"k8s.io/client-go/dynamic"
 )
 
+type initAwaitConfig struct {
```

Comment:
Perhaps add comments on these even though the are internal?  Useful to understand how these are used.

Also - why the word `init` in this struct?  Is that referring to this being the `Create` operation?  If so, perhaps use `create` in the name instead?

Reply:
The thought was that it awaited resource _initialization_ rather than _creation_, where initialization implies that the object has been created _and_ initialized.

But, you're right that it's confusing. Let's change the name and just think of creation as having distinct failure modes -- one for invalid schema errors (_i.e._, the API server rejects the object) and one for initialization errors.